### PR TITLE
docs: add docs about modularizing build components like envdlib

### PR DESCRIPTION
**Pull request by VoVAllen**
Signed-off-by: Jinjing.Zhou <allenzhou@tensorchord.ai>

### CODE REVIEW

**Comment by aseaday** [comment]
LGTM

**Review by aseaday — APPROVED** [review]

File: docs/guide/modularize.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Modularize your build function
+
+Envd allows you to easily create reusable components, so that the building file can be more organized.
```
Comment (xieydd) [inline docs/guide/modularize.md:3]: envd keep lowercase : )

File: docs/guide/modularize.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Defining your own package
+
+envd support most Python syntax, including string format, for-loop, if-statement, function definition. 
+
+For example, you can define a function for tensorflow package like this
+```python
+def tensorfloow(version):
```
Comment (xieydd) [inline docs/guide/modularize.md:11]: tensorfloow -> tensorflow

File: docs/guide/modularize.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+```
+The full list of language spec can be found at https://github.com/google/starlark-go/blob/master/doc/spec.md.
+
+::: tip
+More examples can be found at https://github.com/tensorchord/envdlib/tree/main/src.
+:::
+
+## Using predefined package by envdlib
+
+envd provided a set of predefined libraries which are commonly used in machine learning tasks. You can find them at https://github.com/tensorchord/envdlib.
```
Comment (xieydd) [inline docs/guide/modularize.md:27]: library catalog  may be discussed in another place.

ML team can build there owner libraries for there field, does it need to be mentioned here?

File: docs/guide/modularize.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+To use it you just need one line in your envd file
+```python
+# import envdlib packages
+envdlib = include("https://github.com/tensorchord/envdlib")
+
+# use it in your build function
+def build():
+    base(os="ubuntu20.04", language="python")
+    envdlib.tensorboard(8888)
```
Comment (terrytangyuan) [inline docs/guide/modularize.md:37]: Is there a way to allow more fine-grained modularization, e.g. `envdlib.monitoring.tensorboard()`?

Comment (kemingy) [reply docs/guide/modularize.md:37]: Currently, it doesn't support that. All the `*.envd` in a git repo will be treated as equivalent. But it's easy to support. Feel free to open an issue if you think it's better.

File: docs/guide/modularize.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+And now you'll have tensorboard on your 8888 port.
```
Comment (kemingy) [inline docs/guide/modularize.md:40]: Need to specify `envdlib.tensorboard(host_port=8888)`. By default, the 1st argument is for envd_port.

**Pull request merged**